batch_job/src/work_queue_factory.c

```diff
@@ -1069,7 +1074,7 @@ int main(int argc, char *argv[])
 		}
 	}	
 
-	if(!project_regex && !(master_host && master_port)) {
+	if(using_catalog < 0) {
```

Comment:
Move this to the if else above:

if(project_regex...)
else if((argc - optind ....
else
  fpritnf(stderr, ....